pkg/auth/middleware.go

```diff
@@ -0,0 +1,118 @@
+package auth
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+var (
+	jwkSet      jwk.Set
+	lastUpdated time.Time
+	jwkMutex    sync.RWMutex
+)
+
+func customHTTPClient() *http.Client {
```

Comment:
can we use options func pattern so its easier to mock and test.
Refer - https://dev.to/kittipat1413/understanding-the-options-pattern-in-go-390c

Reply:
This was removed during refactor for caching.